enclave_manager/avalon_enclave_manager/graphene/graphene_enclave_info.py

```diff
@@ -26,22 +29,24 @@ class GrapheneEnclaveInfo(EnclaveAttributes):
     """
 
 # -------------------------------------------------------------------------
-    def __init__(self, worker_signup_json):
+    def __init__(self, config, worker_signup_json):
         """
         Constructor for GrapheneEnclaveInfo.
         Creates signup object.
         """
+        self._config = config['EnclaveModule']
         self.sealed_data = worker_signup_json["sealed_data"]
         self.verifying_key = worker_signup_json["verifying_key"]
         self.encryption_key = worker_signup_json["encryption_key"]
         self.encryption_key_signature = \
             worker_signup_json["encryption_key_signature"]
         self.enclave_id = self.verifying_key
-        # TODO: Generate IAS AVR
-        self.proof_data = None
+        quote = worker_signup_json["quote"]
+        self.proof_data = self._generate_avr(quote)
```

Comment:
check if `quote` is empty/None before calling `self._generate_avr(quote)`. If quote is empty/None set `self.proof_data = None`

Reply:
Done